packages/bcer-retailer-app/app/src/utils/formatting.tsx

```diff
@@ -15,3 +16,68 @@ export function formatError (error: any) {
+ * whatever field is missing will never validate on submit until after it has been touched at least once due to having a missing key.
+ * @param location Location data from the context
+ * @returns Object:IBusinessLocationValues - a location object that has been converted to be compatable with the edit form
+ */
+export function editLocationFormatting (location: IBusinessLocationValues):IBusinessLocationValues {
+  let formattedLocation = {
+    id: location.id ? location.id : '',
+    addressLine1: location.addressLine1 ? location.addressLine1 : '' ,
+    addressLine2: location.addressLine2 ? location.addressLine2 : '',
+    postal: location.postal ? location.postal : '',
+    city: location.city ? location.city : '',
+    phone: location.phone ? location.phone : '',
+    email: location.email ? location.email : '',
+    underage: '',
+    underage_other: '',
+    health_authority: '',
+    doingBusinessAs: location.doingBusinessAs ? location.doingBusinessAs : '',
+    manufacturing: '',
+    tableData: location.tableData ? location.tableData : {id: undefined},
+    error: location.error ? location.error : undefined
+  }
+
+  const underageMatchString = location.underage.toLowerCase();
+  if (underageMatchString.includes('yes')) {
+    formattedLocation.underage = 'Yes';
+  } else if (underageMatchString.includes('no')) {
+    formattedLocation.underage = 'Yes';
+  } else {
+    formattedLocation.underage = 'other';
+    formattedLocation.underage_other = location.underage_other;
+  } 
+
+  const healthAuthorityMatchString = location.health_authority.toLowerCase();
+  if (healthAuthorityMatchString.includes('viha') || healthAuthorityMatchString.includes('island')) {
+    formattedLocation.health_authority = 'island';
+  } else if (healthAuthorityMatchString.includes('fha') || healthAuthorityMatchString.includes('fraser')) {
+    formattedLocation.health_authority = 'fraser';
+  } else if (healthAuthorityMatchString.includes('iha') || healthAuthorityMatchString.includes('interior')) {
+    // Overlap with VIHA, however all cases of VIHA should get caught first, leaving all others with the string 'iha' to be interior only
+    formattedLocation.health_authority = 'interior';
+  } else if (healthAuthorityMatchString.includes('vcha') || healthAuthorityMatchString.includes('coastal')) {
+    formattedLocation.health_authority = 'coastal';
+  } else if (healthAuthorityMatchString.includes('nha') || healthAuthorityMatchString.includes('northern')) {
+    formattedLocation.health_authority = 'northern';
+  } else {
+    //TODO: for now just put the location.health_authority value back into the formatted object. Eventually once the 'other' option is ready
+    // set formattedLocation.health_authority = 'other' and formattedLocation.health_authority_other = location.health_authority
+    formattedLocation.health_authority = location.health_authority;
+  }
```

Comment:
Maybe we can put these into a utils function somewhere in case we have to re-use this mapping?

```
export const HEALTH_AUTHORITIES_ABBRVS = {
     viha: 'island',
     fha: 'fraser',
    iha: 'interior',
    vcha: 'coastal',
    nha: 'northern'
}
...

formattedLocation.health_authority = HEALTH_AUTHORITIES_ABBRVS[location.health_authority.toLowerCase()];
```

Reply:
unfortunately we can't index a const with this because the requirements are that 'anything with VIHA, Vancouver Island, or Vancouver Island Health Authority' === 'island'. too many possible matches to be able to 1:1 index something

Reply:
Oh wops, that makes sense.

How about something like this?

```
const [abbr, authority] = Object.entries(HEALTH_AUTHORITIES_ABBRVS).find(([k, v]) => 
    healthAuthorityMatchString.includes(k) || healthAuthorityMatchString.includes(v)
) || [];

formattedLocation.health_authority = authority || location.health_authority;
```

Or If anything, maybe we can move the logic into a helper function.
`formattedLocation.health_authority = matchHealthAuthority(location.health_authority.toLowerCase()))`

Reply:
i'd like to leave this expanded as an if/else if block for now, as there is an edge case/the possibility that additional match parameters will be required per 'option'. it's also niche enough that i can pretty much guarantee that this mapping wouldn't be used anywhere else in the app, as it solely applies to 'random text -> formik radio group enum'